README.md

```diff
@@ -1,12 +1,294 @@
-# opensearch-scaling-manager
+# Open-search Scaling Manager
+
+Open Search Simulator is an attempt to mimic to behavior of an AWS on which OpenSearch is deployed.
+
+
+
+### Working Principle of Scaling Manager
+
+------
+
+Scaling manager has following modules
+
+- Fetch Metrics
+- Recommendation
+- Trigger
+- Provision
+- State
+
+![Scaling_Manager_Architecture](https://lucid.app/publicSegments/view/12de2241-e528-4fb2-a891-194ebd2d9c95/image.png)
+
+
+
+**Fetch Metrics:** 
+
+- Scaling Manager  code is deployed on each node available in the cluster where OpenSearch is installed.
+- Only the current master node of the cluster will have the privilege to execute the code.
+- Fetch Metrics collects the monitored metrics of the cluster (Usage of cpu, ram, heap, shard etc.)  and those are indexed into Elasticsearch
+- Old data is purged periodically from the index where the duration can be specified by the user.
+- Collected metrics is next passed to the recommendation module.
+
+
+
+**Recommendation:** 
+
+- Collected metrics are now checked against the rules which are specified by the user in config.yaml file.
+- If the metrics are satisfied against the rules, then it recommends Scale-up-by-1 or Scale-down-by-1.
+- The recommendation data(Scale-up-by-1 or Scale-down-by-1) is maintained in a command queue.
+- Recommended data is next passed to the trigger module.
+
+
+
+**Trigger:** 
+
+- Checks the state of the cluster
```

Comment:
Explain more on trigger and provision module. Explain all the states of provision. Check with @Shrinidhi-P-ML about event based scaling